docs/proposals/proposal-001-trigger-and-deploy.md

```diff
@@ -0,0 +1,233 @@
+
+Participating CNCF projects will deploy their project using a gitops approach
+with flux.
+
+#### Project maintainer triggers pipeline to test a new benchmark
+
+Calling the webhook will trigger the pipeline allowing the changes to be tested.
+
+### Risks and Mitigations
+
+Multiple deployments will produce inaccurate results as we can only accurately
+measure a single project per node. We can set [concurrency](https://docs.github.com/en/actions/using-jobs/using-concurrency)
+in the workflow to ensure only a single instance runs at a time.
+
+Uninstall at end of pipeline fails. We can wait till all flux finalizers are
+removed. In future we could create nodes on demand and delete on completion.
+
+## Design Details
+
+![Pipeline Design](./files/trigger-deploy-pipeline.svg)
+
+### Subscribing to Releases
+
+A YAML file of CNCF projects and any sub components will be stored in the
+tooling repo e.g.
+
+```yaml
+# projects.yaml
+projects:
+  - name: falco
+    sub_components:
+      - ebpf
+      - modern-ebpf
+      - kmod
+```
+
+A scheduled GitHub Action will run every hour and check the Atom feed of
+each project for new releases. To manage the state a GitHub [repository variable](https://docs.github.com/en/actions/learn-github-actions/variables)
+per CNCF project is used to store the latest release version.
+
+If a new release is detected the action will trigger the pipeline for the new
+release and update the variable with the new version. This is to ensure each
+release is only triggered once.
+
+If sub components are specified then the pipeline will be triggered once per
+sub component.
+
+### Trigger
+
+The green reviews pipeline will be triggered by sending a [workflow_dispatch](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event)
+event via the GitHub REST API.
+
+Inputs are
+
+- `cncf_project`: **required** Project to be deployed e.g. `falco`
+- `cncf_project_sub`: **optional** Subcomponent if project has multiple variants
+they wish to test e.g. Falco wish to test 3 falco drivers `modern-ebpf`, `kmod`
+and `ebpf`
+- `version`: **required** Version of project to be tested e.g. `0.37.0`
+
+```sh
+curl -X POST \
+     -H "Accept: application/vnd.github.v3+json" \
+     -H "Authorization: token $GITHUB_PAT" \
+     https://api.github.com/repos/cncf-tags/green-reviews-tooling/actions/workflows/pipeline.yaml/dispatches \
+     -d '{"ref":"0.2.0", "inputs": {"cncf_project": "falco", "cncf_project_sub": "modern-ebpf","version":"0.37.0"}}'
+```
+
+The pipeline is versioned by creating releases of the `green-reviews-tooling`
+repo. The git tag to use is passed via the `ref` param.
+
+The CNCF projects will be given a GitHub fine grained access token limited to
+the `green-reviews-tooling` repo. This token will have
+
+- Read access to code and metadata
+- Read write access to actions
+
+### Deploy
+
+Flux is used to deploy the CNCF project. Projects are able to use either
```

Comment:
@nikimanoledaki Right now the proposal is to use kubectl. I like that from a simplicity POV and because it makes deletion easier via `kubectl delete -f`. 

However the pros you make for using the Flux CLI make sense to me.

In an attempt to reduce cognitive load I would be in favour of aligning how we deploy the projects in prop 1 and the benchmarks in prop 2.

If we're agreed on that I can update the proposal so the pipeline dispatcher workflow calls a falco workflow that deploys using the Flux CLI?

cc @ADiTuri @dipankardas011

Reply:
okay kubectl 👍🏼